units/stats_task_end.go

```diff
@@ -141,11 +129,37 @@ func (j *collectTaskEndDataJob) Run(ctx context.Context) {
 	}
 	j.AddError(err)
 
-	if cloud.IsEc2Provider(j.host.Distro.Provider) && len(j.host.Distro.ProviderSettingsList) > 0 {
-		instanceType, ok := j.host.Distro.ProviderSettingsList[0].Lookup("instance_type").StringValueOK()
-		if ok {
-			msg["instance_type"] = instanceType
+	isHostMode := j.task.IsHostTask()
+	if isHostMode {
+		j.host, err = host.FindOneId(j.RuntimeEnvID)
+		j.AddError(err)
+		if err != nil {
+			return
+		}
+		if j.host == nil {
+			j.AddError(errors.Errorf("finding host '%s'", j.RuntimeEnvID))
+			return
+		}
+		msg["host_id"] = j.host.Id
+		msg["distro"] = j.host.Distro.Id
+		msg["provider"] = j.host.Distro.Provider
+		if cloud.IsEc2Provider(j.host.Distro.Provider) && len(j.host.Distro.ProviderSettingsList) > 0 {
+			instanceType, ok := j.host.Distro.ProviderSettingsList[0].Lookup("instance_type").StringValueOK()
+			if ok {
+				msg["instance_type"] = instanceType
+			}
+		}
+	} else {
+		j.pod, err = pod.FindOneByID(j.RuntimeEnvID)
+		j.AddError(err)
+		if err != nil {
+			return
+		}
+		if j.pod == nil {
+			j.AddError(errors.Errorf("finding pod '%s'", j.RuntimeEnvID))
+			return
 		}
+		msg["pod_id"] = j.pod.ID
```

Comment:
It seems like if we just log the pod ID, we don't need to look up the pod in the DB. Can we add other potentially useful information like the OS, arch, Windows version (if set), CPU, and memory?